runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java

```diff
@@ -139,53 +136,6 @@ public MetricUpdates getUpdates() {
         extractUpdates(counters), extractUpdates(distributions), extractUpdates(gauges));
   }
 
-  /**
-   * @param metricUpdate
-   * @return The MonitoringInfo generated from the metricUpdate.
-   */
-  @Nullable
-  private MonitoringInfo counterUpdateToMonitoringInfo(MetricUpdate<Long> metricUpdate) {
-    SimpleMonitoringInfoBuilder builder = new SimpleMonitoringInfoBuilder(true);
-    MetricName metricName = metricUpdate.getKey().metricName();
-    if (metricName instanceof MonitoringInfoMetricName) {
-      MonitoringInfoMetricName monitoringInfoName = (MonitoringInfoMetricName) metricName;
-      // Represents a specific MonitoringInfo for a specific URN.
-      builder.setUrn(monitoringInfoName.getUrn());
-      for (Entry<String, String> e : monitoringInfoName.getLabels().entrySet()) {
-        builder.setLabel(e.getKey(), e.getValue());
-      }
-    } else { // Note: (metricName instanceof MetricName) is always True.
-      // Represents a user counter.
-      builder.setUrnForUserMetric(
-          metricUpdate.getKey().metricName().getNamespace(),
-          metricUpdate.getKey().metricName().getName());
-      // Drop if the stepname is not set. All user counters must be
-      // defined for a PTransform. They must be defined on a container bound to a step.
-      if (this.stepName == null) {
-        return null;
-      }
-      builder.setPTransformLabel(metricUpdate.getKey().stepName());
-    }
-    builder.setInt64Value(metricUpdate.getUpdate());
-    builder.setTimestampToNow();
-    return builder.build();
-  }
-
-  /** Return the cumulative values for any metrics in this container as MonitoringInfos. */
-  public Iterable<MonitoringInfo> getMonitoringInfos() {
```

Comment:
Please keep this method, just have it call the counter update conversion here, instead of making all the call sites do it.